python/lib/dependabot/python/update_checker/pipenv_version_resolver.rb

```diff
@@ -151,9 +148,18 @@ def handle_pipenv_errors(error)
             raise DependencyFileNotResolvable, msg
           end
 
-          if error.message.include?("Could not find a version") ||
-             error.message.include?("is not a python version")
-            check_original_requirements_resolvable
+          if error.message.match?(PIPENV_RANGE_WARNING)
+            msg = "Pipenv does not support specifying Python ranges "\
+              "(see https://github.com/pypa/pipenv/issues/1050 for more "\
+              "details)."
+            raise DependencyFileNotResolvable, msg
+          end
+
+          check_original_requirements_resolvable if error.message.include?("Could not find a version")
+
+          if error.message.include?("SyntaxError: invalid syntax")
+            raise DependencyFileNotResolvable,
+                  "SyntaxError while installing dependencies. Is one of the dependencies not Python 3 compatible?"
```

Comment:
Wondering if we want to highlight pip has dropped python 2 support

```suggestion
                  "SyntaxError while installing dependencies. Is one of the dependencies not Python 3 compatible?  Pip v21+ has dropped support for Python 2."
```

Reply:
Les, let's do it! I'm looking into upgrading to pip 21.x now as well